backend/src/db/factories.js

```diff
@@ -0,0 +1,227 @@
+      neode.create('Post', buildObject),
+      options.author,
+      options.categories,
+      options.tags,
+    ])
+    await Promise.all([
+      post.relateTo(author, 'author'),
+      Promise.all(categories.map(c => c.relateTo(post, 'post'))),
+      Promise.all(tags.map(t => t.relateTo(post, 'post'))),
+    ])
+    if (buildObject.pinned) {
+      const pinnedBy = await (options.pinnedBy || Factory.build('user', { role: 'admin' }))
+      await pinnedBy.relateTo(post, 'pinned')
+    }
+    return post
+  })
+
+Factory.define('comment')
+  .option('postId', null)
+  .option('post', ['postId'], postId => {
+    if (postId) return neode.find('Post', postId)
+    return Factory.build('post')
+  })
+  .option('authorId', null)
+  .option('author', ['authorId'], authorId => {
+    if (authorId) return neode.find('User', authorId)
+    return Factory.build('user')
+  })
+  .attrs({
+    id: uuid,
+    content: faker.lorem.sentence,
+  })
+  .attr('contentExcerpt', ['contentExcerpt', 'content'], (contentExcerpt, content) => {
+    return contentExcerpt || content
+  })
+  .after(async (buildObject, options) => {
+    const [comment, author, post] = await Promise.all([
+      neode.create('Comment', buildObject),
+      options.author,
+      options.post,
+    ])
+    await Promise.all([comment.relateTo(author, 'author'), comment.relateTo(post, 'post')])
+    return comment
+  })
+
+Factory.define('donations')
+  .attr('id', uuid)
+  .attr('goal', 15000)
+  .attr('progress', 0)
+  .after((buildObject, options) => {
+    return neode.create('Donations', buildObject)
+  })
+
+const emailDefaults = {
+  email: faker.internet.email,
+  verifiedAt: () => new Date().toISOString(),
+}
+
+Factory.define('emailAddress')
+  .attr(emailDefaults)
+  .after((buildObject, options) => {
+    return neode.create('EmailAddress', buildObject)
+  })
+
+Factory.define('unverifiedEmailAddress')
+  .attr(emailDefaults)
+  .after((buildObject, options) => {
+    return neode.create('UnverifiedEmailAddress', buildObject)
+  })
+
+Factory.define('location')
+  .attrs({
+    name: 'Germany',
+    namePT: 'Alemanha',
+    nameDE: 'Deutschland',
+    nameES: 'Alemania',
+    nameNL: 'Duitsland',
+    namePL: 'Niemcy',
+    nameFR: 'Allemagne',
+    nameIT: 'Germania',
+    nameEN: 'Germany',
+    id: 'country.10743216036480410',
+    type: 'country',
+  })
+  .after((buildObject, options) => {
+    return neode.create('Location', buildObject)
+  })
+
+Factory.define('report').after((buildObject, options) => {
+  return neode.create('Report', buildObject)
+})
+
+Factory.define('tag')
+  .attrs({
+    name: '#human-connection',
+  })
+  .after((buildObject, options) => {
+    return neode.create('Tag', buildObject)
+  })
+
+Factory.define('socialMedia')
+  .attrs({
+    url: 'https://mastodon.social/@Gargron',
+  })
+  .after((buildObject, options) => {
+    return neode.create('SocialMedia', buildObject)
+  })
+
+export default Factory
```

Comment:
out of curiosity, why did  you decide to collate this all in one file instead of having separate files for each like before?

Reply:
I'm open for changes here. It's because this one file exports `Factory` which needs to go through all the factory definitions.